src/utils/validator-extras.ts

```diff
@@ -0,0 +1,133 @@
+import dayjs from 'dayjs';
+import forEach from 'lodash/forEach';
+import origValidator from 'validator';
+import type { Attributes, Model } from '../model.js';
+
+type OrigValidator = typeof origValidator;
+
+// extend validator methods
+export interface Extensions {
+  extend(name: string, fn: (...args: unknown[]) => unknown): unknown;
+  // Blocked by: https://github.com/microsoft/TypeScript/issues/7765
+  // NEED_SUGGESTION: How to properly create a custom validation?
```

Comment:
Might be good to migrate the following test to TS to test this;
https://github.com/sequelize/sequelize/blob/dc21c5c3bc9dfca628cea8dca5eaeb844cda1593/test/integration/instance.validations.test.js#L704-L727